powerauth-nextstep-client/src/main/java/io/getlime/security/powerauth/lib/nextstep/client/NextStepClient.java

```diff
@@ -199,6 +200,88 @@ private RestTemplate defaultTemplate() {
     }
 
     /**
+     * Get all authentication methods supported by Next Step server.
+     *
+     * @return List of authentication methods wrapped in GetAuthMethodResponse.
+     * @throws NextStepServiceException Exception with {@link ErrorModel} for ERROR status
+     */
+    public Response<GetAuthMethodsResponse> getAuthMethods() throws NextStepServiceException {
+        try {
+            // Exchange next step request with NextStep server.
+            ResponseEntity<Response<GetAuthMethodsResponse>> response = defaultTemplate().exchange(serviceUrl + "/auth-method", HttpMethod.GET, null, new ParameterizedTypeReference<Response<GetAuthMethodsResponse>>() {
+            });
+            return new Response<>(Response.Status.OK, response.getBody().getResponseObject());
+        } catch (HttpStatusCodeException ex) {
+            throw handleHttpError(ex);
+        } catch (ResourceAccessException ex) {
+            throw handleResourceAccessError(ex);
+        }
+    }
+
+    /**
+     * Get all enabled authentication methods for given user.
+     *
+     * @param userId User ID
+     * @return List of enabled authentication methods for given user wrapped in GetAuthMethodResponse.
+     * @throws NextStepServiceException Exception with {@link ErrorModel} for ERROR status
+     */
+    public Response<GetAuthMethodsResponse> getAuthMethodsEnabledForUser(String userId) throws NextStepServiceException {
+        try {
+            // Exchange next step request with NextStep server.
+            ResponseEntity<Response<GetAuthMethodsResponse>> response = defaultTemplate().exchange(serviceUrl + "/user/" + userId + "/auth-method", HttpMethod.GET, null, new ParameterizedTypeReference<Response<GetAuthMethodsResponse>>() {
+            });
+            return new Response<>(Response.Status.OK, response.getBody().getResponseObject());
+        } catch (HttpStatusCodeException ex) {
+            throw handleHttpError(ex);
+        } catch (ResourceAccessException ex) {
+            throw handleResourceAccessError(ex);
+        }
+    }
+
+    /**
+     * Enable an authentication method for given user.
+     *
+     * @param userId     User ID
+     * @param authMethod Authentication method
+     * @return List of enabled authentication methods for given user wrapped in GetAuthMethodResponse.
+     * @throws NextStepServiceException Exception with {@link ErrorModel} for ERROR status
+     */
+    public Response<GetAuthMethodsResponse> enableAuthMethodForUser(String userId, AuthMethod authMethod) throws NextStepServiceException {
+        try {
+            // Exchange next step request with NextStep server.
+            ResponseEntity<Response<GetAuthMethodsResponse>> response = defaultTemplate().exchange(serviceUrl + "/user/" + userId + "/auth-method/" + authMethod, HttpMethod.POST, null, new ParameterizedTypeReference<Response<GetAuthMethodsResponse>>() {
```

Comment:
Do not include userId in URL, please use request object to carry userId attribute.